account/identity.go

```diff
@@ -58,11 +58,17 @@ func (u NullUUID) Value() (driver.Value, error) {
 // One User account can have many Identities
 type Identity struct {
 	gormsupport.Lifecycle
-	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field. For identities provided by Keyclaok this ID equals to the Keycloak. For other types of IDP (github, oso, etc) this ID is generated automaticaly
-	Username string    // The username of the Identity
-	Provider string    // The identity provider ID, such as "keycloak", "github", "oso", etc
-	UserID   NullUUID  `sql:"type:uuid"` // Belongs to User
-	User     User
+	// This is the ID PK field. For identities provided by Keyclaok this ID equals to the Keycloak. For other types of IDP (github, oso, etc) this ID is generated automaticaly
+	ID uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	// The username of the Identity
+	Username string
+	// ProviderType The type of provider, such as "keycloak", "github", "oso", etc
+	ProviderType string `gorm:"column:provider_type"`
```

Comment:
I agree with you that `provider` vs `providerType` may not make a big difference when dealing with GitHub because there's a single github.com, but when we consider remote items synchronized from a JIRA service (JBoss, Hibernate, etc.), then I think that the term `providerType` becomes more accurate.